### DISCUSSION TITLE
mosquitto.conf example on_connect

### DISCUSSION THREAD

**Discussion by dac2020** [arendst/Tasmota#16989] [Q&A] [answered]
I know this is not related to Tasmota specifically but someone here might know.
I read that using the on_connect directive in the broker conf can show auth errors in the logs.

However, I've searched for two days and find only programming mentions of this directive and not any cases where someone is using it as just that, a directive. 

I had found one mention somewhere saying that the directive needed four values.
http://www.steves-internet-guide.com/mqtt-username-password-example/

His example seems to show this but this seems to be being used in python.
on_connect(client, userdata, flags, rc):

In mosquitto documentation, best I can find is to use the connection_messages [ true | false ] directive to log connection info.

Does anyone know how to log authentication error attempts so we can know when there are auth errors.

**Comment by sfromis** [comment, marked as answer]
What Steves Internet Guide tells about `on_connect` has nothing to do with `mosquitto.conf`, but when coding a client in Python using the paho library.

The docs page about `mosquitto.conf` does have a lot if info, including in relation to the logging it can do.
https://mosquitto.org/man/mosquitto-conf-5.html

**Reply by dac2020** [reply]
Yes, I was looking at the doc page but wondered if maybe I was missing something.
Thanks for confirming.
